### PR TITLE
Fixing issue #74 - hread View is not enforcing LoginToRead

### PR DESCRIPTION
**Pull request by bentruby**

### CODE REVIEW

File: user/showthread.php [hunk 1]
```diff
@@ -6,6 +6,14 @@
...
 require_once("page-yatt.inc.php");
 
+if(isset($forum['option']['LoginToRead']) and $forum['option']['LoginToRead']) {
+  $user->req();
+  if ($user->status != 'Active') {
+    echo "Your account isn't validated\n";
+    exit;
+  }
+}
+
```
Comment (nyetwurk) [inline user/showthread.php:16]: I see this code block copy/pasted between a bunch of different files; maybe a single common function would make sense.

Comment (nyetwurk) [reply user/showthread.php:16]: Here is another variant in showtracking.php
```
if (!$user->valid()) header("Location: /login.phtml?url=$url");
```

Comment (bentruby) [reply user/showthread.php:16]: Either would work, what I submitted has been implemented in a production environment without issue.

Comment (nyetwurk) [reply user/showthread.php:16]: Will merge this for now, and will open a new ticket for making common code.

**Review by nyetwurk — APPROVED** [review]

**Pull request merged**